iam-login-service/src/main/java/it/infn/mw/iam/api/aup/model/AupDTO.java

```diff
@@ -22,16 +22,20 @@
 import javax.validation.constraints.Size;
 
 import org.hibernate.validator.constraints.NotBlank;
+import org.hibernate.validator.constraints.URL;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 
 import it.infn.mw.iam.api.scim.controller.utils.JsonDateSerializer;
+import it.infn.mw.iam.api.validators.HtmlEscapeCheck;
 
 public class AupDTO {
 
-  @NotBlank(message = "Invalid AUP: the AUP text cannot be blank")
-  String text;
+  @NotBlank(message = "Invalid AUP: the AUP URL cannot be blank")
+  @URL
+  @HtmlEscapeCheck
```

Comment:
We will probably need to be more strict here, and forbid query parameters at all.